### PR TITLE
Accounts can be listed, activated, and deactivated

### PR DESCRIPTION
**Pull request by zspencer**
Adds Account Management API features  to 4.0

### CODE REVIEW

File: lib/nylas/types.rb [hunk 1]
```diff
@@ -79,5 +79,14 @@ def cast(value)
       end
     end
     Types.registry[:string] = StringType.new
+
+    # Type for attributes represented as booleans.
+    class BooleanType < ValueType
+      # @param value [Object] Strictly casts the passed in value to a boolean (must be true, not "" or 1)
+      def cast(value)
```
Comment (zspencer) [inline lib/nylas/types.rb]: I am wondering if this is nil, it should be nil?

That way if we get a nil value in something that could be boolean, we don't assume it's false. Thoughts @khamidou?

File: lib/nylas/account.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    attribute :id, :string
+    attribute :account_id, :string
+    attribute :billing_state, :string
+    attribute :sync_state, :string
+
+    attribute :email, :string
+    attribute :trial, :boolean
+
+    def activate
```
Comment (khamidou) [inline lib/nylas/account.rb]: Hmm, I think we should keep the same terminology as our existing API – why not rename these methods `upgrade!` and `downgrade!`?

File: lib/nylas/http_client.rb [hunk 3]
```diff
@@ -111,11 +117,11 @@ def default_headers
     rescue JSON::ParserError
       response
     end
+    inform_on :parse_response, level: :debug, also_log: { result: true }
```
Comment (khamidou) [inline lib/nylas/http_client.rb]: Curious, what does `inform_on` do?

File: lib/nylas/model.rb [hunk 4]
```diff
@@ -18,22 +18,27 @@ def self.included(model)
     def save
       raise_if_read_only
       result = if id
-                 api.execute(method: :put, payload: attributes.serialize, path: resource_path)
+                 exec
```
Comment (khamidou) [inline lib/nylas/model.rb]: Hmm, did you mean to leave `exec` here?

File: lib/nylas/types.rb (hunk 1 above)
Comment (khamidou) [reply lib/nylas/types.rb]: Hmm, I don't think of a case where this would happen. Maybe we should raise an exception?

Comment (zspencer) [reply lib/nylas/types.rb]: I can see it in more generic use cases, but not with your API as your API doesn't send non-true-values. Exception makes sense!

File: lib/nylas/http_client.rb (hunk 3 above)
Comment (zspencer) [reply lib/nylas/http_client.rb]: inform_on logs the entire output of the `parse_response` if you have your NYLAS_LOG_LEVEL set to `debug` or greater verbosity. 

Very useful for diagnosis and exploration.

File: lib/nylas/model.rb (hunk 4 above)
Comment (zspencer) [reply lib/nylas/model.rb]: ah, good catch! It's a no-op since it doesn't do anything, but leaving a system call lying out is not a good thing!

**Pull request merged**